### ISSUE TITLE
Bottleneck retrained model predicts one class after applying transforms and opimize_for_inference. 

### ISSUE THREAD

**Issue by h3nok** [tensorflow/tensorflow#19235]
Summary : 

**Have I written custom code** - Mostly using the retrain script from the tensorflow for poets tutorial. I added custom code to validate network but didn't change the training procedure. 
**OS Platform and Distribution** -  Windows 7 x64. Training and evaluation is done on a windows machine using Anconda python distribution. Once I freeze the model I then copy it to a linux box with CentOS 7 x86_64 to do the transformation. The transformed model is then copied back to the windows box for further analysis before deployment.
**TensorFlow installed from** - anaconda cloud 
**TensorFlow version** - 1.2.1 
**Bazel version** - 0.8.1
**CUDA/cuDNN version** - 8 / 7 
**GPU model and memory** - GTX 970 4GB 
**Exact command to reproduce** - N/A

I have asked the same question on stackoverflow 

https://stackoverflow.com/questions/50243492/inception-v3-retrained-model-incorrect-predictions-after-using-optimize-for-infe/50263837#50263837

and opencv help 

http://answers.opencv.org/question/191168/inception-v3-retrained-model-incorrect-predictions-after-using-optimize_for_inference-and-graph_transform-tools/

I can really use some suggestions: 

1. I followed the tensorflow for poets tutorial to retrain inception V3 for 3-class image task 

https://codelabs.developers.google.com/codelabs/tensorflow-for-poets/#0

2. The network fits the data well and i'm able to validate the frozen model using tensorflow python which predicts unseen samples at 97 % accuracy 

3. The frozen model has to be integrated to a legacy application which uses opencv::DNN module. To get the forzen model ready I applied 
  - optimize_for_inference 
  - graph_transforms with " remove_nodes(op=PlaceholderWithDefault) strip_unused_nodes(type=float, shape=\"1,299,299,3\") sort_by_execution_order" flags 

4. Loaded transformed model with opencv and run some more validation before deployment. The transformed version could not match (not even close) the trained original model.  It mostly predicts a single class. 

I used opencv blobFromImage function with swapRB set to true. 

I tried to normalize input by subtracting ImageNet mean as well as my training set mean. Unsuccessful in both cases. 

I just have no idea what is causing for the model to be so different after its transformed. 

Thanks for your time

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Have I written custom code
OS Platform and Distribution
TensorFlow installed from
TensorFlow version
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce

**Comment by h3nok** [comment]
Updated question with platform etc info

**Comment by h3nok** [comment]
This was primarily due to the preprocessing and batch normalization that happens during training.To normalize an input at inference time please use 127.5 for RGB mean values and set the scale factor to 1/127.5 when calling opencv::dnn::blobFromImage. You might want to swap RB channels depending how you did the training.
